libr/core/cmd_api.c

```diff
@@ -185,114 +186,251 @@ R_API RCmdDesc *r_cmd_get_desc(RCmd *cmd, const char *cmd_identifier) {
+	int bytes_from_first = first->is_str
+		? first->sz - 1
+		: first->sz;
+
+	out_sz = bytes_from_first + second_sz;
+	out = malloc (out_sz);
+
+	/* Copy full buffer if raw bytes. Stop before \0 if string. */
+	memcpy (out, first->data, bytes_from_first);
+	/* Always copy all bytes from second, including trailing \0 */
+	memcpy (out+bytes_from_first, second, second_sz);
+
+	if (out_sz) {
+		*out_szp = out_sz;
+	}
+	return out;
+}
+
+R_API int r_cmd_alias_append_str(RCmd *cmd, const char *k, const char *a) {
+	RCmdAliasVal *v_old = r_cmd_alias_get (cmd, k);
+	if (v_old) {
+		if (!v_old->is_data) {
+#if DEBUG_ALIAS
+			eprintf ("Won't append to existing command alias.\n");
+#endif
 			return 1;
 		}
+
+		int new_len;
+		ut8* new = alias_append_internal (&new_len, v_old, (ut8 *)a, strlen (a) + 1);
+
+		r_cmd_alias_set_raw (cmd, k, new, new_len);
+		free (new);
+#if DEBUG_ALIAS
+		eprintf ("Appended alias.\n");
+#endif
+	} else {
+		r_cmd_alias_set_str (cmd, k, a);
+#if DEBUG_ALIAS
+		eprintf ("Creating new alias while trying to append.\n");
+#endif
 	}
 
-	i = cmd->aliases.count++;
-	char **K = (char **)realloc (cmd->aliases.keys,
-				     sizeof (char *) * cmd->aliases.count);
-	if (K) {
-		cmd->aliases.keys = K;
-		int *R = (int *)realloc (cmd->aliases.remote,
-				sizeof (int) * cmd->aliases.count);
-		if (R) {
-			cmd->aliases.remote = R;
-			char **V = (char **)realloc (cmd->aliases.values,
-					sizeof (char *) * cmd->aliases.count);
-			if (V) {
-				cmd->aliases.values = V;
-				cmd->aliases.keys[i] = strdup (k);
-				cmd->aliases.values[i] = strdup (v);
-				cmd->aliases.remote[i] = remote;
-			}
+	return 0;
+}
+
+R_API int r_cmd_alias_append_raw(RCmd *cmd, const char *k, const ut8 *a, int sz) {
+	RCmdAliasVal *v_old = r_cmd_alias_get (cmd, k);
+	if (v_old) {
+		if (!v_old->is_data) {
+#if DEBUG_ALIAS
+			eprintf ("Won't append to existing command alias.\n");
+#endif
+			return 1;
 		}
+
+		int new_len;
+		ut8 *new = alias_append_internal (&new_len, v_old, a, sz);
+
+		r_cmd_alias_set_raw (cmd, k, new, new_len);
+		free (new);
+#if DEBUG_ALIAS
+		eprintf ("Appended alias.\n");
+#endif
+	} else {
+		r_cmd_alias_set_raw (cmd, k, a, sz);
+#if DEBUG_ALIAS
+		eprintf ("Creating new alias while trying to append.\n");
+#endif
 	}
-	free (tofree);
+
 	return 0;
 }
 
-R_API const char *r_cmd_alias_get(RCmd *cmd, const char *k, int remote) {
-	r_return_val_if_fail (cmd && k, NULL);
-	int matches, i;
-	// TODO: use a hashtable
-	for (i = 0; i < cmd->aliases.count; i++) {
-		matches = 0;
-		if (remote) {
-			if (cmd->aliases.remote[i]) {
-				matches = !strncmp (k, cmd->aliases.keys[i],
-					strlen (cmd->aliases.keys[i]));
-			}
-		} else {
-			matches = !strcmp (k, cmd->aliases.keys[i]);
-		}
-		if (matches) {
-			return cmd->aliases.values[i];
-		}
+/* Returns a new copy of v->data. If !v->is_str, hex escaped */
+R_API char *r_cmd_alias_val_strdup(RCmdAliasVal *v) {
+	if (v->is_str) {
+		return strdup ((char *)v->data);
+	} else {
```

Comment:
Unnecessary else

Reply:
Will remove, though imo explicit else is fine in cases like this